src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -760,9 +767,48 @@ private protected override void SaveModel(ModelSaveContext ctx)
             for (int i = 0; i < _ldas.Length; i++)
             {
                 _ldas[i].Save(ctx);
+
+                if(_columns[i].OutputTopicWordSummary)
+                    SaveTopicWordSummary(ctx, i);
             }
         }
 
+        private void SaveTopicWordSummary(ModelSaveContext ctx, int i)
+        {
+            var summary = GetLdaDetails(i);
+
+            ctx.SaveTextStream(WordTopicModelFilename, writer =>
```

Comment:
I know that the `WordTopicModelFilename` was unused in the codebase before your changes on this PR, and it was a legacy variable coming from TLC (as commented [here](https://github.com/dotnet/machinelearning/issues/4322#issuecomment-572333759)), but I'm not sure this is the best way to use it.

The potential problem I see is that for every _columns[i] that we save, we'll call this method (if _columns[i].OutputTopicWordSummary == true, as seen in your code above), and all of them will try to save their WordSummary to the same `WordTopicModelFilename`.

So if a given LdaTransformer has more than 1 column with `OutputTopicWordSummary` enabled, then 2 or more columns will try to output to the same file.

Looking at the implementation of `ctx.SaveTextStream` I'm not sure if it will throw if we try to write twice the same file. If it does, this would be a problem. If it doesn't throw, there's still the problem that we will have the output topics of all the columns in one single file, and no way to know which file is of which column.

So adding text to identify where the output of one column begins would be useful. Another sugestion would be to actually use one file per column (for example, something like ""word_topic_summary-col{i}.txt" might be sufficient.

(PS: I don't know what was the behavior back on TLC, but maybe there we couldn't have 1 Lda transformer acting on several columns).

Reply:
sounds good, I will add column option name as part of summary text file

---
In reply to: [445988583](https://github.com/dotnet/machinelearning/pull/5260#discussion_r445988583) [](ancestors = 445988583)